hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -588,6 +588,19 @@ protected void runTableServicesInline(HoodieTable table, HoodieCommitMetadata me
         metadata.addMetadata(HoodieClusteringConfig.SCHEDULE_INLINE_CLUSTERING.key(), "true");
         inlineScheduleClustering(extraMetadata);
       }
+
+      // if clustering is disabled, but we might need to rollback any inflight clustering when clustering was enabled previously.
+      if (!config.inlineClusteringEnabled() && !config.isAsyncClusteringEnabled() && !config.scheduleInlineClustering()
+          && config.isRollbackPendingClustering()) {
+        // rollback any pending clustering
+        table.getActiveTimeline().filterPendingReplaceTimeline().getInstants().forEach(instant -> {
+          Option<Pair<HoodieInstant, HoodieClusteringPlan>> instantPlan = ClusteringUtils.getClusteringPlan(table.getMetaClient(), instant);
+          if (instantPlan.isPresent()) {
+            LOG.info("Rolling back pending clustering at instant " + instantPlan.get().getLeft());
+            rollback(instant.getTimestamp());
```

Comment:
This API ignores the pending rollbacks for clustering.  For the following case, does it leave an inflight rollback in the timeline indefinitely?

timeline: commit1, commit2, replacecommit3.requested (clustering), rollback1.inflight (rolling back replacecommit3, job failed in the middle)

Reply:
don't we have this issue even today. may be worth fixing it separately for all.

Reply:
https://issues.apache.org/jira/browse/HUDI-5169

Reply:
This issue exists today, but before this PR, there should not be a pending rollback for a clustering instant, so it was not a problem.  Only when this PR introduces the rollback of a clustering instant, it becomes an issue, and the metadata table can still be blocked by such a pending rollback.  So that should be fixed in this PR without introducing new problems.

Reply:
actually, rollback() takes care of ensureing if there is already a rollback targetting the commit of interest, it will re-use the same. we can sync up directly.